app/scripts/services/membership/membership.js

```diff
@@ -0,0 +1,172 @@
+'use strict';
+
+angular
+  .module('openshiftConsole')
+  .factory('MembershipService', function($filter) {
+
+    var annotation = $filter('annotation');
+
+    var userRoles = function(roleBindings, user) {
+                      return _.map(
+                              _.filter(roleBindings, function(binding) {
+                                return _.find(binding.subjects, function(subject) {
+                                  return subject.name === user.metadata.name;
+                                });
+                              }),
+                              'metadata.name');
+    };
+
+    var isLastRole = function(userName, roleBindings) {
+      return  _.isEqual(
```

Comment:
Using lodash helpers for a simple array length check is actually making this code harder to parse.  Similar to the above change, if you are checking for the existence of at least one thing in an array, use _.some
```
var userRoles = _.filter(roleBindings, function(binding) {
  return _.some(binding.subjects, {name: userName});
});
return userRoles.length === 1;
```